### ISSUE TITLE
Required fields in form are still required even if field/group is hidden

### ISSUE THREAD

**Issue by jardakotesovec** [pkp/pkp-lib#9996]
**Describe the bug**
Orcid form has use case, where first field enables Orcid and as result display rest of the form and some of the fields there are required. Problem is during validation, because the validation does not take in account if some fields are hidden.

**Proposed solution**
I think that improving form behaviour so it skips required fields if they are conditionally hidden would be expected behaviour.

**What application are you using?**
 OJS, OMP or OPS version main

**Comment by jardakotesovec** [comment]
@ewhanson This should resolve the workaround you are having in OrcidForm.vue. What you think?

ui-library: https://github.com/pkp/ui-library/pull/357

**Comment by ewhanson** [comment]
Hey @jardakotesovec, thanks for this! I think this gets most of the way to solving my issues. It is possible to save a form with conditionally hidden required elements blank, but if there are previous errors, they are not dismissed when the fields are hidden (see video). 

https://github.com/pkp/pkp-lib/assets/20844845/4402b824-291e-41a1-b1d9-4842e3c0eaa3

**Comment by jardakotesovec** [comment]
@ewhanson Here is second attempt. Got quick idea here - just to remove the error when the field is unmounting (being removed/hidden), which can be think about as clean up for field which is not relevant anymore. 

At this point I think this should work well..  can't think of where it could cause problems at this point.. but if you can think of something please let me know!

**Comment by ewhanson** [comment]
Thanks @jardakotesovec! This is most of the way there now, but I'm still running into an error. It seems that only one error is removed at a time when the components are unmounted. I have an example where there are 3 "required fields not completed" errors. And each time I hide/unmount the components, another one of the errors is removed.

https://github.com/pkp/pkp-lib/assets/20844845/7b7d6b11-c114-43d7-931e-bf903c45dd36

**Comment by jardakotesovec** [comment]
@ewhanson Great catch, thanks! I did not test it with more than one field, so I did not notice it.

It went wrong, because how the state management is structured currently there. So every removing error created new errors object with that one error removed. But the errors object prop don't get propagated between these calls.. so all of them were modifying the original object and basically last change win.

Not fan of the solution I end up with.. hence the long comment in the code..  but it does work and I still think that this is how it should behave and it should remove the errors if the field gets removed.
